py/modbuiltins.c

```diff
@@ -440,8 +440,18 @@ STATIC mp_obj_t mp_builtin_round(mp_uint_t n_args, const mp_obj_t *args) {
     mp_int_t num_dig = 0;
     if (n_args > 1) {
         num_dig = mp_obj_get_int(args[1]);
-        if (num_dig > 0) {
-            mp_not_implemented("round(..., N>0)");
+        if (num_dig > 0 && num_dig < 7) {
+            mp_float_t val = mp_obj_get_float(o_in);
+            mp_float_t mult = MICROPY_FLOAT_C_FUN(pow)(10, num_dig);
+            mp_float_t rounded = MICROPY_FLOAT_C_FUN(round)(val * mult) / mult;
+            return mp_obj_new_float(rounded);
+        } else if (num_dig > 0 && num_dig > 6) {
+            mp_not_implemented("round(..., N>6)");
+        } else if (num_dig < 0) {
+            mp_float_t val = mp_obj_get_float(o_in);
+            mp_float_t mult = MICROPY_FLOAT_C_FUN(pow)(10, MICROPY_FLOAT_C_FUN(fabs)(num_dig));
+            mp_float_t rounded = MICROPY_FLOAT_C_FUN(round)(val * mult);
```

Comment:
> will remove it

Well, so far it was just my own vote (well, @danicampora now agrees too). Feel free to wait for other reviewers. Ideal solution would be to see how much code size <0 case adds and then decide.